app/controllers/InterstitialController.scala

```diff
@@ -95,13 +96,33 @@ class InterstitialController @Inject() (
   private def currentUrl(implicit request: Request[AnyContent]) =
     configDecorator.pertaxFrontendHost + request.path
 
-  def displayChildBenefits: Action[AnyContent] = authenticate { implicit request =>
-    Ok(
-      viewChildBenefitsSummaryInterstitialView(
-        redirectUrl = currentUrl,
-        taxCreditsEnabled = configDecorator.taxCreditsEnabled
-      )
-    )
+  def displayChildBenefits: Action[AnyContent] = authenticate.async { implicit request =>
+    featureFlagService.get(ChildBenefitSingleAccountToggle).map { featureFlag =>
+      if (!featureFlag.isEnabled) {
+        Ok(
+          viewChildBenefitsSummaryInterstitialView(
+            redirectUrl = currentUrl,
+            taxCreditsEnabled = configDecorator.taxCreditsEnabled
+          )
+        )
+      } else {
+        errorRenderer.error(UNAUTHORIZED)
```

Comment:
Can you redirect to the new page using a 301 status.
Remember that urls get bookmarked or can be used by other services. Always send the user to the new location is the url changes.